tests/test_polynomials_repr.py

```diff
@@ -0,0 +1,107 @@
+"""Unit-testing module defining polynomials __str__ test cases."""
+
+import unittest
+from polynomial import (
+    Constant,
+    LinearBinomial,
+    Monomial,
+    Polynomial,
+    QuadraticTrinomial,
+    ZeroPolynomial,
+)
+
+
+class TestPolynomialsRepr(unittest.TestCase):
+    """Defines polynomials __str__ test cases."""
+
+    def test_smoke(self):
+        """Perform a test case containing all possible complicated inputs."""
+        coeffs = [0, 0, -2, 0, 4.6666666666, -(69 + 420j), -1, 1337]
+        expect = "Polynomial(-2, 0, 4.6666666666, (-69-420j), -1, 1337)"
+
+        r = repr(Polynomial(coeffs))
+
+        self.assertEqual(expect, r)
+
+    def test_negative_coefficient_unchanged(self):
+        """Test that negative coefficients replace the + between the terms."""
+        coeffs = [3, -2, 0, 0]
+        expect = "Polynomial(3, -2, 0, 0)"
+
+        r = repr(Polynomial(coeffs))
+
+        self.assertEqual(expect, r)
+
+    def test_trailing_zero_kept(self):
+        """Test that the first-degree-term ends with 'x' and not 'x^1'."""
+        coeffs = [2, 0]
+        expect = "Polynomial(2, 0)"
+
+        r = repr(Polynomial(coeffs))
+
+        self.assertEqual(r, expect)
+
+    def test_constant_kept(self):
+        """Test that the constant term does not end with 'x' or 'x^0'."""
+        coeffs = [2]
+        expect = "Polynomial(2)"
+
+        r = repr(Polynomial(coeffs))
+
+        self.assertEqual(expect, r)
+
+    def test_internal_zero_kept(self):
+        """Test that constants with value 1 are displayed properly."""
+        coeffs = [1, 0, 1]
+        expect = "Polynomial(1, 0, 1)"
+
+        r = repr(Polynomial(coeffs))
+
+        self.assertEqual(expect, r)
```

Comment:
I think that these few test cases now are redundant. Testing `__str__` has all these specific things that might come up as bugs but there is nothing like that with `__repr__` (talking about `test_negative_coefficient_unchanged`,  `test_trailing_zero_kept`, `test_constant_kept` and `test_internal_zero_kept`). I can imagine at most testing `repr(Polynomial(0, 0, 0))` or something like that as an edge case but I think the mentioned tests above are totally unnecessary and irrelevant as is the case now with `__repr__`'s implementation.